src/core-sdk-tasks/DownloadFile.cs

```diff
@@ -39,31 +54,70 @@ public override bool Execute()
                 var filePath = Uri.Substring(FileUriProtocol.Length);
                 Log.LogMessage($"Copying '{filePath}' to '{DestinationPath}'");
                 File.Copy(filePath, DestinationPath);
+                return true;
             }
-            else
+
+            var lastStatus = await DownloadWithRetries(Uri, DestinationPath);
+            if (lastStatus == false)
             {
-                Log.LogMessage(MessageImportance.High, $"Downloading '{Uri}' to '{DestinationPath}'");
+                if (!string.IsNullOrEmpty(PrivateUri))
+                {
+                    lastStatus = await DownloadWithRetries(PrivateUri, DestinationPath);
+                }
+            }
+
+            return lastStatus == true;
+        }
 
+        /// <summary>
+        /// Attempt to download file from `source` with retries when response error is different of FileNotFound and Success.
+        /// </summary>
+        /// <param name="source">URL to the file to be downloaded.</param>
+        /// <param name="target">Local path where to put the downloaded file.</param>
+        /// <returns>true: Download Succeeded. false: Download failed with 404. null: Download failed but is retriable.</returns>
+        private async Task<bool?> DownloadWithRetries(string source, string target)
```

Comment:
Missing async suffix:

```suggestion
        private async Task<bool?> DownloadWithRetriesAsync(string source, string target)
```